discord/ext/commands/bot.py

```diff
@@ -39,21 +45,34 @@
 from .help import HelpCommand, DefaultHelpCommand
 from .cog import Cog
 
+if TYPE_CHECKING:
+    import importlib.machinery
+    
+    from ...message import Message
+    from ._types import (
+        Check,
+        CoroFunc,
+    )
+
 __all__ = (
     'when_mentioned',
     'when_mentioned_or',
     'Bot',
     'AutoShardedBot',
 )
 
-def when_mentioned(bot, msg):
+T = TypeVar('T')
+CFT = TypeVar('CFT', bound='CoroFunc')
+CXT = TypeVar('CXT', bound='Context')
```

Comment:
Do we want a proper naming convention set for typevars?

Reply:
A convention I've seen used that I think is helpful is `CoroFuncT` and `ContextT`